covalent_dispatcher/_dal/result.py

```diff
@@ -0,0 +1,525 @@
+
+    def get_value(self, key: str, session: Session = None, refresh: bool = True):
+        return get_filters[key](super().get_value(key, session, refresh))
+
+    def set_value(self, key: str, val: Any, session: Session = None) -> None:
+        super().set_value(key, set_filters[key](val), session)
+
+    def _update_dispatch(
+        self,
+        start_time: datetime = None,
+        end_time: datetime = None,
+        status: "Status" = None,
+        error: str = None,
+        result: Any = None,
+    ):
+        """
+        Update the dispatch metadata.
+
+        Args:
+            start_time: The start time of the lattice execution.
+            end_time: The end time of the lattice execution.
+            status: The status of the lattice execution.
+            result: The lattice output unless error occured in which case None.
+            error: Any error that occurred
+
+        """
+
+        with self.session() as session:
+            if start_time is not None:
+                self.set_value("start_time", start_time, session)
+            if end_time is not None:
+                self.set_value("end_time", end_time, session)
+            if status is not None:
+                self.set_value("status", status, session)
+            if error is not None:
+                self.set_value("error", error, session)
+            if result is not None:
+                self.set_value("result", result, session)
+
+        # Copy output and error assets to sublattice's parent electron
+        if RESULT_STATUS.is_terminal(status) and self._electron_id:
+            with self.session() as session:
+                electron_rec = Electron.get_db_records(
+                    session,
+                    keys={"id", "parent_lattice_id"},
+                    equality_filters={"id": self._electron_id},
+                    membership_filters={},
+                )[0]
+                parent_electron = Electron(session, electron_rec)
+
+                subl_output = self.get_asset("result", session)
+                subl_err = self.get_asset("error", session)
+                electron_output = parent_electron.get_asset("output", session)
+                electron_err = parent_electron.get_asset("error", session)
+
+            app_log.debug("Copying sublattice output to parent electron")
+            with self.session() as session:
+                copy_asset_meta(session, subl_output, electron_output)
+                copy_asset_meta(session, subl_err, electron_err)
+
+            copy_asset(subl_output, electron_output)
+            copy_asset(subl_err, electron_err)
+
+    def _update_node(
+        self,
+        node_id: int,
+        node_name: str = None,
+        start_time: datetime = None,
+        end_time: datetime = None,
+        status: "Status" = None,
+        output: Any = None,
+        error: Exception = None,
+        stdout: str = None,
+        stderr: str = None,
+        qelectron_data_exists: bool = False,
```

Comment:
@kessler-frost It would be better to default this to `None` . Each electron will receive more than one status update throughout its lifecycle and not all parties will intend to update the electron's `qelectron_data_exists` attribute or even be aware of qelectrons.